core/src/mender-api.c

```diff
@@ -38,7 +38,7 @@
  * @brief Paths of the mender-server APIs
  */
 #define MENDER_API_PATH_POST_AUTHENTICATION_REQUESTS "/api/devices/v1/authentication/auth_requests"
-#define MENDER_API_PATH_GET_NEXT_DEPLOYMENT          "/api/devices/v1/deployments/device/deployments/next"
+#define MENDER_API_PATH_GET_NEXT_DEPLOYMENT          "/api/devices/v2/deployments/device/deployments/next"
```

Comment:
@joelguittet Please keep in mind that v2 API (and v1 POST) are Enterprise-only endpoints. It works on HM but in won't work on an Open Source server.

There are three API end-points for the client to get an update:
* v1 GET -> uses only artifact_name
* v1 POST -> uses devices specified "provides"
* v2 POST -> uses devices specified "provides" and supports update control maps (which is a deprecated feature, by the way)

In the Linux client we implement this as a fallback, so the client tries the Enterprise endpoint first and on errors tries the basic v1 GET.

Reply:
This is a major information you provide here, so I realize I should not have merge the PR too much quickly. This detail is not provided at https://docs.mender.io/api/#device-api-deployments-v2, I suggest an enhancement of the doc should be done on Mender side to indicate the v2 api is only valid for Enterprise!

Reply:
The nicely rendered APIs at https://docs.mender.io/api are for the Enterprise flavors of each service. To see the open source APIs you need to go directly into each microservice documentation, for example https://github.com/mendersoftware/deployments/blob/master/docs/devices_api.yml